simplified-main/src/main/java/org/nypl/simplified/main/MainFragmentViewModel.kt

```diff
@@ -11,13 +29,150 @@ class MainFragmentViewModel : ViewModel() {
 
   private val logger = LoggerFactory.getLogger(MainFragmentViewModel::class.java)
 
-  /**
-   * `true` if the history of tabs should be cleared.
-   */
+  private val disposables = mutableListOf<Disposable>()
+  private val services = Services.serviceDirectory()
+  private val accountProviders by lazy {
+    this.services.requireService(AccountProviderRegistryType::class.java)
+  }
+  private val profilesController by lazy {
+    this.services.requireService(ProfilesControllerType::class.java)
+  }
+  private val crashlytics by lazy {
+    services.optionalService(CrashlyticsServiceType::class.java)
+  }
+
+  init {
+    this.disposables.add(
+      profilesController.accountEvents()
+        .subscribe(this::onAccountEvent)
+    )
+    this.disposables.add(
+      profilesController.profileEvents()
+        .subscribe(this::onProfileEvent)
+    )
+  }
+
+  val currentProfile: ProfileReadableType
+    get() {
+      return this.profilesController.profileCurrent()
+    }
+  val currentAccount: AccountReadableType?
+    get() {
+      return this.currentProfile.mostRecentAccount() ?: when (this.currentProfile.accounts().size) {
+        0 -> throw UnreachableCodeException() // We expect one account to always exist
+        1 -> this.currentProfile.accounts().values.first()
+        else -> {
+          val defaultProvider = this.accountProviders.defaultProvider
+          this.currentProfile.accounts().values.first {
+            it.provider.id != defaultProvider.id
+          }
+        }
+      }
+    }
+
+  override fun onCleared() {
+    super.onCleared()
+
+    // Clear any disposables before the view model is destroyed
+    disposables.forEach { it.dispose() }
```

Comment:
With the `CompositeDisposable`, this then just becomes `disposables.dispose()`

Reply:
Nice. That's cleaner.